### ISSUE TITLE
Oracle: SELECT...INTO with record/collection-field targets (e.g. \INTO rec.field`) is unparsable, causing false ST03 unused-CTE positives`

### ISSUE THREAD

**Issue by Ollec** [sqlfluff/sqlfluff#8250]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Oracle `PL/SQL SELECT ... INTO` clauses that target fields of a record variable (e.g. `INTO out_rec.col1, out_rec.col2`) fail to parse. The `.field` portion after the first `INTO `identifier; and everything following it in the statement, including the trailing `FROM `clause; collapses into a single `unparsable` segment. As a knock-on effect, because the `FROM` `cte_name` text is now buried inside `unparsable` rather than a real from_clause/table_reference node, rule ST03 (`structure.unused_cte`) incorrectly reports the referenced **CTE** as unused, since its usage-detection (`recursive_crawl("table_reference")`) never sees a proper `table_reference` node for it.

### Expected Behaviour

The `INTO` clause should parse cleanly when its targets are dotted record/collection-field references (e.g. `out_rec.col1`), not just bare identifiers or bind variables.

No `PRS` (parse) error, and no `ST03` violation for `cte1`, since it's genuinely referenced in the `FROM` clause.

### Observed Behaviour

See "How to reproduce" for repro.sql lines.

```
== [repro.sql] FAIL
L:   5 | P:   8 | ST03 | Query defines CTE "cte1" but does not use it.
                       | [structure.unused_cte]
L:  12 | P:  16 |  PRS | Line 12, Position 16: Found unparsable section: '.a,\n
                       | out_metrics.b\n  from cte1'
WARNING: Parsing errors found and dialect is set to 'oracle'. Have you configured your dialect correctly?
```
`sqlfluff parse` shows the break clearly; parsing succeeds up through the first `INTO` target (`out_metrics` as a bare `naked_identifier`), then hits unparsable right at the following `.`:

```
into_clause:
    keyword:                  'into'
    newline:                  '\n'
    naked_identifier:         'out_metrics'
unparsable:                   !! Expected: 'Nothing here.'
    dot:                      '.'
    word:                     'a'
    comma:                    ','
    ...
    word:                     'from'
    word:                     'cte1'
```

### How to reproduce

Create a file called repro.sql with the following lines:
```
create or replace procedure test_proc(out_rec out sys_refcursor) is
  type t_rec is record (a number, b number);
  out_metrics t_rec;
begin
  with cte1 as (
    select 1 as x from dual
  )
  select
    cte1.x,
    cte1.x
  into
    out_metrics.a,
    out_metrics.b
  from cte1;
end;
/
```

`sqlfluff lint --dialect oracle --rules ST03 repro.sql`
`sqlfluff parse --dialect oracle repro.sql`

### Dialect

oracle

### Version

sqlfluff, version 4.2.0

### Configuration

No project .sqlfluff/.sqlfluffignore needed to reproduce; the repro above fails with just --dialect oracle passed on the CLI, no config file present. 

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
